cmd/bootstrap/transit/cmd/utils.go

```diff
@@ -0,0 +1,289 @@
+	}
+
+	// create the destination file
+	destination, err := os.Create(dst)
+	if err != nil {
+		return fmt.Errorf("failed to create file %s: %w", dst, err)
+	}
+	defer destination.Close()
+
+	// copy the source file to the destination file
+	_, err = io.Copy(destination, source)
+	if err != nil {
+		errorStr := err.Error()
+		closeErr := source.Close()
+		if closeErr != nil {
+			errorStr = fmt.Sprintf("%s, %s", errorStr, closeErr)
+		}
+		return fmt.Errorf("failed to copy file %s to %s: %s", src, dst, errorStr)
+	}
+
+	// close the source file
+	err = source.Close()
+	if err != nil {
+		return fmt.Errorf("failed to close source file %s: %w", src, err)
+	}
+
+	// flush the destination file
+	err = destination.Sync()
+	if err != nil {
+		return fmt.Errorf("failed to copy file %s to %s: %w", src, dst, err)
+	}
+
+	// read the source file permissions
+	si, err := os.Stat(src)
+	if err != nil {
+		return fmt.Errorf("failed to get file information %s: %w", src, err)
+	}
+
+	// set the same permissions on the destination file
+	err = os.Chmod(dst, si.Mode())
+	if err != nil {
+		return fmt.Errorf("failed to set permisson on file %s: %w", dst, err)
+	}
+
+	// delete the source file
+	err = os.Remove(src)
+	if err != nil {
+		return fmt.Errorf("failed removing original file: %s", err)
+	}
+
+	return nil
+}
+
+func unWrapFile(nodeID string) error {
+
+	log.Info().Msg("decrypting Random Beacon key")
+
+	pubKeyPath := filepath.Join(flagBootDir, fmt.Sprintf(FilenameTransitKeyPub, nodeID))
+	privKeyPath := filepath.Join(flagBootDir, fmt.Sprintf(FilenameTransitKeyPriv, nodeID))
+	ciphertextPath := filepath.Join(flagBootDir, fmt.Sprintf(FilenameRandomBeaconCipher, nodeID))
+	plaintextPath := filepath.Join(flagBootDir, fmt.Sprintf(bootstrap.PathRandomBeaconPriv, nodeID))
+
+	ciphertext, err := ioutils.ReadFile(ciphertextPath)
+	if err != nil {
+		return fmt.Errorf("Failed to open ciphertext file %s: %w", ciphertextPath, err)
+	}
+
+	publicKey, err := ioutils.ReadFile(pubKeyPath)
+	if err != nil {
+		return fmt.Errorf("Failed to open public keyfile %s: %w", pubKeyPath, err)
+	}
+
+	privateKey, err := ioutils.ReadFile(privKeyPath)
+	if err != nil {
+		return fmt.Errorf("Failed to open private keyfile %s: %w", privKeyPath, err)
+	}
+
+	// NaCl is picky and wants its type to be exactly a [32]byte, but readfile reads a slice
+	var pubKeyBytes, privKeyBytes [32]byte
+	copy(pubKeyBytes[:], publicKey)
+	copy(privKeyBytes[:], privateKey)
+
+	plaintext := make([]byte, 0, len(ciphertext)-box.AnonymousOverhead)
+	plaintext, ok := box.OpenAnonymous(plaintext, ciphertext, &pubKeyBytes, &privKeyBytes)
+	if !ok {
+		return fmt.Errorf("Failed to decrypt random beacon key using private key from file: %s", privKeyPath)
+	}
+
+	err = ioutil.WriteFile(plaintextPath, plaintext, fileMode)
+	if err != nil {
+		return fmt.Errorf("Failed to write the decrypted file %s: %w", plaintextPath, err)
+	}
+
+	return nil
+}
+
+// func wrapFile(nodeID string) error {
```

Comment:
Why is this commented out? Was it not previously used?

I see it used [here](https://github.com/onflow/flow-go/pull/553/files#diff-25f9f5d376044c3d8ad19f5e758d6b960a15d9afab22beccb4e1586fd964921bL96-L103). Is this replicated in the new command structure?

Reply:
I've added a command `wrap` to deal with this, would this be desirable or some other form?

Reply:
A separate command seems like the right way to handle this 👍